pkg/apis/pipeline/v1beta1/taskrun_validation.go

```diff
@@ -93,6 +94,22 @@ func (ts *TaskRunSpec) Validate(ctx context.Context) (errs *apis.FieldError) {
 	return errs
 }
 
+// validateStepComputeResources validates compute resources are not configured in TaskRunSpec.Step
+func validateStepComputeResources(computeResources *corev1.ResourceRequirements, taskSpec *TaskSpec) (errs *apis.FieldError) {
```

Comment:
Suggested name/docstring:

// validateComputeResources validates that compute resources specified for the TaskRun are compatible with the Task spec, i.e. that compute resources are not specified at both the Task level (on the TaskRun) and Step level (on the Task)

Also, this function assumes that it is only called if taskRun.computeResources is not nil, but I think it would be better if this function were self contained.

Reply:
I see. the `taskRun.computeResources` check could be in the func to more explicitly show the validation. 
thanks this.